fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1517,6 +1533,159 @@ public Void visitDefaultValueExpr(DefaultValueExpr node, Scope context) {
         public Void visitCloneExpr(CloneExpr node, Scope context) {
             return null;
         }
+
+        @Override
+        public Void visitDictQueryExpr(DictQueryExpr node, Scope context) {
+            List<Expr> params = node.getParams().exprs();
+            if (!(params.get(0) instanceof StringLiteral)) {
+                throw new SemanticException("dict_mapping function first param table_name should be string literal");
+            }
+            String[] dictTableFullName = ((StringLiteral) params.get(0)).getStringValue().split("\\.");
+            TableName tableName;
+            if (dictTableFullName.length == 1) {
+                tableName = new TableName(null, dictTableFullName[0]);
+                tableName.normalization(session);
+            } else if (dictTableFullName.length == 2) {
+                tableName = new TableName(dictTableFullName[0], dictTableFullName[1]);
+            } else {
+                throw new SemanticException("dict_mapping function first param table_name should be 'db.tbl' or 'tbl' format");
+            }
+
+            Database db = GlobalStateMgr.getCurrentState().getDb(tableName.getDb());
+            if (db == null) {
+                throw new SemanticException("Database %s is not found", tableName.getDb());
+            }
+            Table table = db.getTable(tableName.getTbl());
+            if (table == null) {
+                throw new SemanticException("dict table %s is not found", tableName.getTbl());
+            }
+            if (!(table instanceof OlapTable)) {
+                throw new SemanticException("dict table type is not OlapTable, type=" + table.getClass());
+            }
+            if (table instanceof MaterializedView) {
+                throw new SemanticException("dict table can't be materialized view");
+            }
+            OlapTable dictTable = (OlapTable) table;
+
+            if (dictTable.getKeysType() != KeysType.PRIMARY_KEYS) {
+                throw new SemanticException("dict table " + tableName + " should be primary key table");
+            }
+
+            // verify keys length and type
+            List<Column> keyColumns = new ArrayList<>();
+            Column valueColumn = null;
+            for (Column column : dictTable.getBaseSchema()) {
+                if (column.isKey()) {
+                    keyColumns.add(column);
+                }
+                if (column.isAutoIncrement()) {
+                    valueColumn = column;
+                }
+            }
+            // (table, keys..., value_column, strict_mode)
```

Comment:
Suppose pk column is string, string, boolean, if we call dict_mapping("dict_name", "pk_cols_name", "value_column", strict_mode) it can be success in this case? If does, how can we avoid this problem?

Reply:
dict table must be pk table, and in params check can know pk columns size
dict_mapping(''dict_table', keys [, value_column] [, strict_mode])
keys length = pk size, value_column is string, strict_mode is boolean
and this function doesn't support implicit cast to avoid ambiguity.